src/ServiceControl/Recoverability/Archiving/ArchiveDocumentManager.cs

```diff
@@ -38,15 +38,10 @@ public ArchiveOperation CreateArchiveOperation(IDocumentSession session, string
             int documentCount = 0;
             var indexQuery = session.Query<FailureGroupMessageView>(new FailedMessages_ByGroup().IndexName);
 
-            if (cutOff.HasValue)
```

Comment:
What are the consequences of this removal?

Reply:
It means we no longer wait for non stale results as of that cutoff etag. It seems like they've removed this API completely in raven 4, but it [should still be there in 3.5](https://ravendb.net/docs/article-page/3.5/Csharp/client-api/session/querying/how-to-customize-query#waitfornonstaleresultsasof). The cutoff was the current time when the archive all messages message was created.

Perhaps we should re-include it or move closer to the v4 version by switching to [waitfornonstaleresults](https://ravendb.net/docs/article-page/3.5/Csharp/client-api/session/querying/how-to-customize-query#waitfornonstaleresults)

Reply:
Unfortunately we can't do it. The reason why I removed it was this runtime exceptions:

```
System.NotSupportedException: 'Since Stream() does not wait for indexing (by design), streaming query with WaitForNonStaleResults is not supported.'
```

Reply:
Damn meaning let's merge it?